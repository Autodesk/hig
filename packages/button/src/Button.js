import React, { Component } from "react";
import PropTypes from "prop-types";
import { ControlBehavior } from "@hig/behaviors";

import {
  availableTargets,
  availableTypes,
  availableWidths,
  types,
  widths
} from "./constants";
import ButtonPresenter from "./presenters/ButtonPresenter";

export default class Button extends Component {
  static propTypes = {
    /**
     * Prevents user interaction with the button
     */
    disabled: PropTypes.bool,
    /**
     * A @hig/icon element
     */
    icon: PropTypes.node,
    /**
     * Sets the link of a button
     */
    link: PropTypes.string,
    /**
     * Triggers blur when focus is moved away from icon
     */
    onBlur: PropTypes.func,
    /**
     * Triggers when you click the button
     */
    onClick: PropTypes.func,
    /**
     * Triggers when focus is moved to button
     */
    onFocus: PropTypes.func,
    /**
     * Triggers when you hover over the button
     */
    onHover: PropTypes.func,
    /**
     * Triggers when the user's mouse is pressed over the button
     */
    onMouseDown: PropTypes.func,
    /**
     * Triggers when the user's mouse is over the button
     */
    onMouseEnter: PropTypes.func,
    /**
     * Triggers when the user's mouse is no longer over the button
     */
    onMouseLeave: PropTypes.func,
    /**
     * Triggers when the user's mouse is no longer pressed over the button
     */
    onMouseUp: PropTypes.func,
    /**
     * Adds custom/overriding styles
     */
    stylesheet: PropTypes.func,
    /**
     * Specifies where to display the linked URL
     */
    target: PropTypes.oneOf(availableTargets),
    /**
     * Sets the title of a button
     */
    title: PropTypes.string.isRequired,
    /**
     * Specifies type of button
     */
    type: PropTypes.oneOf(availableTypes),
    /**
     * Specifies width of button
     */
    width: PropTypes.oneOf(availableWidths)
  };

  static defaultProps = {
    disabled: false,
    type: types.SOLID,
    width: widths.SHRINK
  };

  render() {
    const {
      disabled,
      icon,
      link,
      onBlur,
      onClick,
      onFocus,
      onHover,
      onMouseDown,
      onMouseEnter,
      onMouseLeave,
      onMouseUp,
      stylesheet,
      target,
      title,
      type,
      width,
      ...otherProps
    } = this.props;

    return (
      <ControlBehavior
        onBlur={onBlur}
        onFocus={onFocus}
        onMouseDown={onMouseDown}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
        onMouseUp={onMouseUp}
      >
        {({
          hasFocus,
          hasHover,
          isPressed,
          onBlur: handleBlur,
          onFocus: handleFocus,
          onMouseDown: handleMouseDown,
          onMouseEnter: handleMouseEnter,
          onMouseLeave: handleMouseLeave,
          onMouseUp: handleMouseUp
        }) => (
          <ButtonPresenter
            disabled={disabled}
            hasFocus={hasFocus}
            hasHover={hasHover}
            isPressed={isPressed}
            icon={icon}
            link={link}
            onBlur={handleBlur}
            onClick={onClick}
            onFocus={handleFocus}
            onHover={onHover}
            onMouseDown={handleMouseDown}
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
            onMouseUp={handleMouseUp}
            stylesheet={stylesheet}
            target={target}
            title={title}
            type={type}
            width={width}
            {...otherProps}
          />
        )}
      </ControlBehavior>
    );
  }
}
