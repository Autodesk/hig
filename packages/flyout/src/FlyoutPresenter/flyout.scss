@import "~@hig/styles/mixins/colors.scss";
@import "~@hig/styles/mixins/shadows.scss";
@import "~@hig/styles/tokens/layers.scss";

.hig__flyout-v1 {
  position: relative;
  display: inline-block;
}

.hig__flyout-v1__container {
  @import "./tokens.scss";

  position: absolute;
  // Resolves issues with negative positions and contaier overflow
  display: table;
  width: 100%;
  z-index: $flyout-container-layer;
  transition-property: opacity, transform;
  transition-duration: $container-transition-duration;
  // Bounce effect
  transition-timing-function: $container-transition-easing;

  .hig__flyout-v1--exited & {
    width: 0;
    height: 0;
    overflow: hidden;
    touch-action: none;
    pointer-events: none;
  }
}

//  [Container Transition Styles]
//
//  This block contains all container transition styles in a nested format.
//  Due to the many different combinations of modifiers in the Flyout,
//  the use of nested styles allows for much more succinct declarations.

.hig__flyout-v1--exiting,
.hig__flyout-v1--exited {
  @import "./tokens.scss";

  .hig__flyout-v1__container {
    opacity: 0;
  }

  &.hig__flyout-v1--top-left,
  &.hig__flyout-v1--top-center,
  &.hig__flyout-v1--top-right {
    .hig__flyout-v1__container {
      transform: translateY(-$container-transition-distance);
    }
  }

  &.hig__flyout-v1--right-top,
  &.hig__flyout-v1--right-center,
  &.hig__flyout-v1--right-bottom {
    .hig__flyout-v1__container {
      transform: translateX($container-transition-distance);
    }
  }

  &.hig__flyout-v1--bottom-left,
  &.hig__flyout-v1--bottom-center,
  &.hig__flyout-v1--bottom-right {
    .hig__flyout-v1__container {
      transform: translateY($container-transition-distance);
    }
  }

  &.hig__flyout-v1--left-top,
  &.hig__flyout-v1--left-center,
  &.hig__flyout-v1--left-bottom {
    .hig__flyout-v1__container {
      transform: translateX(-$container-transition-distance);
    }
  }
}

.hig__flyout-v1__panel {
  @import "./tokens.scss";
  @include shadow("common");

  box-sizing: border-box;
  min-width: 200px;
  max-width: 400px;
  max-height: 360px;
  padding: $panel-padding;
  overflow-y: auto;
  -ms-overflow-style: -ms-autohiding-scrollbar;
  border: $panel-border-width solid color(hig-cool-gray-20);
  background-color: $panel-background-color;
  border-radius: $panel-border-radius;
}

.hig__flyout-v1__chevron {
  @import "./tokens.scss";

  position: absolute;
  width: $chevron-size;
  height: $chevron-size;
  background: $panel-background-color;
  transform: rotate(45deg);

  .hig__flyout-v1--top-left &,
  .hig__flyout-v1--top-center &,
  .hig__flyout-v1--top-right & {
    box-shadow: (-$chevron-shadow-offset) (-$chevron-shadow-offset)
      $chevron-shadow-color;
  }

  .hig__flyout-v1--right-top &,
  .hig__flyout-v1--right-center &,
  .hig__flyout-v1--right-bottom & {
    box-shadow: $chevron-shadow-offset (-$chevron-shadow-offset)
      $chevron-shadow-color;
  }

  .hig__flyout-v1--bottom-left &,
  .hig__flyout-v1--bottom-center &,
  .hig__flyout-v1--bottom-right & {
    box-shadow: $chevron-shadow-offset $chevron-shadow-offset
      $chevron-shadow-color;
  }

  .hig__flyout-v1--left-top &,
  .hig__flyout-v1--left-center &,
  .hig__flyout-v1--left-bottom & {
    box-shadow: (-$chevron-shadow-offset) $chevron-shadow-offset
      $chevron-shadow-color;
  }

  .hig__flyout-v1--top-left & {
    top: $chevron-side-position;
    left: $chevron-spacing;
  }

  .hig__flyout-v1--top-center & {
    top: $chevron-side-position;
    left: $chevron-centering-position;
  }

  .hig__flyout-v1--top-right & {
    top: $chevron-side-position;
    right: $chevron-spacing;
  }

  .hig__flyout-v1--right-top & {
    right: $chevron-side-position;
    top: $chevron-spacing;
  }

  .hig__flyout-v1--right-center & {
    right: $chevron-side-position;
    top: $chevron-centering-position;
  }

  .hig__flyout-v1--right-bottom & {
    right: $chevron-side-position;
    bottom: $chevron-spacing;
  }

  .hig__flyout-v1--bottom-left & {
    bottom: $chevron-side-position;
    left: $chevron-spacing;
  }

  .hig__flyout-v1--bottom-center & {
    bottom: $chevron-side-position;
    left: $chevron-centering-position;
  }

  .hig__flyout-v1--bottom-right & {
    bottom: $chevron-side-position;
    right: $chevron-spacing;
  }

  .hig__flyout-v1--left-top & {
    left: $chevron-side-position;
    top: $chevron-spacing;
  }

  .hig__flyout-v1--left-center & {
    left: $chevron-side-position;
    top: $chevron-centering-position;
  }

  .hig__flyout-v1--left-bottom & {
    left: $chevron-side-position;
    bottom: $chevron-spacing;
  }
}

.hig__flyout-v1__container::after {
  @import "./tokens.scss";

  content: "";
  position: absolute;
  display: block;
  background: $panel-background-color;
  border-radius: $panel-border-radius;

  .hig__flyout-v1--top-left &,
  .hig__flyout-v1--top-center &,
  .hig__flyout-v1--top-right & {
    top: $panel-border-width;
    left: $panel-border-width;
    right: $panel-border-width;
    height: $panel-padding;
  }

  .hig__flyout-v1--right-top &,
  .hig__flyout-v1--right-center &,
  .hig__flyout-v1--right-bottom & {
    top: $panel-border-width;
    right: $panel-border-width;
    bottom: $panel-border-width;
    width: $panel-padding;
  }

  .hig__flyout-v1--bottom-left &,
  .hig__flyout-v1--bottom-center &,
  .hig__flyout-v1--bottom-right & {
    bottom: $panel-border-width;
    left: $panel-border-width;
    right: $panel-border-width;
    height: $panel-padding;
  }

  .hig__flyout-v1--left-top &,
  .hig__flyout-v1--left-center &,
  .hig__flyout-v1--left-bottom & {
    top: $panel-border-width;
    bottom: $panel-border-width;
    left: $panel-border-width;
    width: $panel-padding;
  }
}
