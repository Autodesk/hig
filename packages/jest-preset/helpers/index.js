/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
