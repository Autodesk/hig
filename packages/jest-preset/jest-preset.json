{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
