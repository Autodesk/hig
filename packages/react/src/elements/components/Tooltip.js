import * as PropTypes from "prop-types";
import React, { Component } from "react";
import * as HIG from "hig-vanilla";
import TooltipAdapter from "../../adapters/TooltipAdapter";

class Tooltip extends Component {
  static propTypes = {
    anchorPoint: PropTypes.oneOf(HIG.Tooltip.AvailableAnchorPoints),
    children: PropTypes.node,
    content: PropTypes.string
  };

  static __docgenInfo = {
    props: {
      anchorPoint: {
        description: "where the tooltip will be anchored relative to target"
      },
      children: { description: "target component to open the tooltip" },
      content: { description: "content for the tooltip" }
    }
  };

  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }

  closeTooltip = () => {
    this.setState({ open: false });
  };

  openTooltip = () => {
    this.setState({ open: true });
  };

  render() {
    const target = this.props.children
      ? React.cloneElement(this.props.children, { onClick: this.openTooltip })
      : null;
    return (
      <TooltipAdapter
        anchorPoint={this.props.anchorPoint}
        open={this.state.open}
        onClickOutside={this.closeTooltip}
        content={this.props.content}
      >
        {target}
      </TooltipAdapter>
    );
  }
}

export default Tooltip;
