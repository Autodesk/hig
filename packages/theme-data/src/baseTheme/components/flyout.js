import { BORDER_RADIUS, COLOR, LENGTH } from "../../consts/types";

export default {
  "flyout.backgroundColor": {
    type: COLOR,
    value: { ref: "colorScheme.surface.level100" }
  },
  "flyout.borderColor": {
    type: COLOR,
    value: { ref: "colorScheme.opacity.transparent" }
  },
  "flyout.borderRadius": {
    type: BORDER_RADIUS,
    value: { ref: "basics.borderRadii.large" }
  },
  "flyout.shadowBlur": {
    type: LENGTH,
    value: { ref: "basics.shadows.highBlur" }
  },
  "flyout.shadowColor": {
    type: COLOR,
    value: { ref: "colorScheme.shadow.high" }
  },
  "flyout.nub.hypotenuseLength": {
    type: LENGTH,
    value: "16px"
  }
};
