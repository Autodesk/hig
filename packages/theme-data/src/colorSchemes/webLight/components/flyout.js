export default {
  "flyout.borderColor": {
    value: "#d4dbe1"
  }
};
