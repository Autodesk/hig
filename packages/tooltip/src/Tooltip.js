import React from "react";
import PropTypes from "prop-types";
import Flyout, {
  dislocateContainer,
  AVAILABLE_ANCHOR_POINTS
} from "@hig/flyout";

import PanelPresenter from "./presenters/PanelPresenter";
import PointerPresenter from "./presenters/PointerPresenter";

export default function Tooltip(props) {
  return <Flyout {...props} />;
}

Tooltip.defaultProps = {
  /**
   * @param {import("@hig/flyout").Coordinates} coordinates
   * @returns {import("@hig/flyout").Coordinates}
   */
  alterCoordinates(coordinates) {
    return dislocateContainer(coordinates, 10);
  },
  /**
   * @param {import("@hig/flyout").PanelRendererPayload} payload
   */
  /* eslint-disable-next-line react/prop-types */
  panel({ innerRef, content, handleScroll, maxHeight }) {
    return (
      <PanelPresenter
        innerRef={innerRef}
        maxHeight={maxHeight}
        onScroll={handleScroll}
      >
        {content}
      </PanelPresenter>
    );
  },
  pointer: <PointerPresenter />
};

/**
 * This should be the same as `Flyout`,
 * but we need to copypasta since docgen
 * doesn't understand references.
 */
Tooltip.propTypes = {
  /** Manipulate flyout coordinates before rendering */
  alterCoordinates: PropTypes.func,
  /** Where the flyout will be anchored relative to target */
  anchorPoint: PropTypes.oneOf(AVAILABLE_ANCHOR_POINTS),
  /** Target component to open the flyout. Can be either a node or a render function */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  /** Content for the flyout. Can be either a node or a render function */
  content: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
  /** Default uncontrolled open state */
  defaultOpen: PropTypes.bool,
  /**
   * When the flyout overflows the viewport, it'll attempt to
   * use the given anchor points in order to keep the flyout
   * within the viewport.
   */
  fallbackAnchorPoints: PropTypes.arrayOf(
    PropTypes.oneOf(AVAILABLE_ANCHOR_POINTS)
  ),
  /** Use to render a custom flyout panel. Can be either a node or a render function */
  panel: PropTypes.func,
  /** A custom pointer */
  pointer: PropTypes.node,
  /** Max height of the flyout content, in pixels */
  maxHeight: PropTypes.number,
  /** Function called when the flyout is open, and a click event occurs outside the flyout */
  onClickOutside: PropTypes.func,
  /** Function called when the flyout closes */
  onClose: PropTypes.func,
  /** Function called when the flyout opens */
  onOpen: PropTypes.func,
  /** Function called when the flyout panel is scrolled */
  onScroll: PropTypes.func,
  /** When provided, it overrides the flyout's open state */
  open: PropTypes.bool,
  /** Whether flyout should open when the target is hovered over */
  openOnHover: PropTypes.bool,
  /**
   * If openOnHover is true, this prop will determine the delay
   * from when mouseEnter begins until the Tooltip visually opens
   */
  openOnHoverDelay: PropTypes.number
};
