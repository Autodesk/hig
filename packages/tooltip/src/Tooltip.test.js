import { anchorPoints } from "@weave-design/flyout";
import Button from "@weave-design/button";
import React from "react";
import { takeSnapshotsOf } from "@weave-design/jest-preset/helpers";
import Tooltip from "./Tooltip";

describe("tooltip/Tooltip", () => {
  takeSnapshotsOf(Tooltip, [
    {
      desc: "renders without props",
      props: {},
    },
    {
      desc: "renders with props",
      props: {
        anchorPoint: anchorPoints.TOP_CENTER,
        children: <Button title="Click Me" />,
        content: "Hello World",
        maxHeight: 150,
        onClickOutside: function onClickOutside() {},
        onScroll: function onScroll() {},
        open: true,
        openOnHover: false,
      },
    },
  ]);
});
