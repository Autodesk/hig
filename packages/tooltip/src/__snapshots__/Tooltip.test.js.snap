// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip/Tooltip  1`] = `
.emotion-7 {
  position: relative;
  display: inline-block;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  position: absolute;
  display: none;
  width: 100%;
  z-index: 9999;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  touch-action: auto;
  pointer-events: auto;
  opacity: 1;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-1 {
  fill: #535353;
}

.emotion-5 {
  border: 1px solid #535353;
  background-color: #535353;
  border-radius: 2px;
}

.emotion-4 {
  position: relative;
}

<div
  className="emotion-7"
>
  <div
    className="emotion-0"
  />
  <div
    className="emotion-6"
    style={
      Object {
        "left": "0px",
        "top": "0px",
      }
    }
  >
    <div
      aria-hidden="true"
      className="emotion-3"
      role="presentation"
      style={
        Object {
          "left": "0px",
          "top": "0px",
        }
      }
    >
      <svg
        height="12"
        viewBox="0 0 12 12"
        width="12"
      >
        <polygon
          className="emotion-1"
          points="0,6 6,0 12,6"
        />
        <polygon
          className="emotion-1"
          points="2.5,6 6,2.5 9.5,6"
        />
      </svg>
    </div>
    <div
      className="emotion-5"
      onScroll={undefined}
      style={
        Object {
          "maxHeight": undefined,
        }
      }
    >
      <div
        className="emotion-4"
      />
    </div>
  </div>
</div>
`;

exports[`tooltip/Tooltip  2`] = `
.emotion-8 {
  position: relative;
  display: inline-block;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.emotion-2 {
  fill: #535353;
}

.emotion-6 {
  border: 1px solid #535353;
  background-color: #535353;
  border-radius: 2px;
}

.emotion-5 {
  position: relative;
}

.emotion-7 {
  position: absolute;
  display: table;
  width: 100%;
  z-index: 9999;
  -webkit-transition-property: opacity,-webkit-transform;
  -webkit-transition-property: opacity,transform;
  transition-property: opacity,transform;
  -webkit-transition-duration: 250ms;
  transition-duration: 250ms;
  -webkit-transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  transition-timing-function: cubic-bezier(0.77,0,0.265,2);
  touch-action: auto;
  pointer-events: auto;
  opacity: 1;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-0 {
  border-width: 1px;
  border-style: solid;
  border-color: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: ArtifaktElement,sans-serif;
  font-size: 14px;
  font-weight: 600;
  line-height: 18px;
  margin: 0;
  overflow: hidden;
  padding: 8px 16px;
  position: relative;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-overflow: ellipsis;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-transition-property: box-shadow;
  transition-property: box-shadow;
  -webkit-transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  transition-timing-function: cubic-bezier(0.4,0,0.2,1);
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  background: #0696d7;
}

.emotion-0:visited {
  color: #ffffff;
}

.emotion-0 svg * {
  fill: #ffffff;
}

<div
  className="emotion-8"
>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-0"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={undefined}
      onMouseUp={[Function]}
      tabIndex="0"
      target={undefined}
    >
      <span
        className=""
      >
        Click Me
      </span>
    </button>
  </div>
  <div
    className="emotion-7"
    style={
      Object {
        "left": "0px",
        "top": "0px",
      }
    }
  >
    <div
      aria-hidden="true"
      className="emotion-4"
      role="presentation"
      style={
        Object {
          "left": "0px",
          "top": "0px",
        }
      }
    >
      <svg
        height="12"
        viewBox="0 0 12 12"
        width="12"
      >
        <polygon
          className="emotion-2"
          points="0,6 6,0 12,6"
        />
        <polygon
          className="emotion-2"
          points="2.5,6 6,2.5 9.5,6"
        />
      </svg>
    </div>
    <div
      className="emotion-6"
      onScroll={[Function]}
      style={
        Object {
          "maxHeight": "150px",
        }
      }
    >
      <div
        className="emotion-5"
      >
        Hello World
      </div>
    </div>
  </div>
</div>
`;
